### ISSUE TITLE
versionCode must be increased with each release

### ISSUE THREAD

**Issue by IzzySoft** [freshfieldreds/muffed#4]
I've just noticed you forgot to increase `versionCode`:

    package: name='com.freshfieldreds.muffed' versionCode='1' versionName='0.2.0'

`versionCode` is what Android uses internally to tell versions apart and decide whether an APK is an update to the installed app – so without increasing it, updates won't be signaled to users. A "force update" is still possible (as long as `versionCode` was not decreased), but then the user must be aware that it exists. So please increase `versionCode` whenever you prepare a new release. The place for that, with Flutter, is usually [here](https://github.com/freshfieldreds/muffed/blob/main/pubspec.yaml#L6):

```yaml
version: 0.2.0+3
```

would be the correct notation (I've chosen `3` as this is your third release – you can increase as much as you want, just make sure the number following the `+` is an integer and higher than it was before).

Thanks for caring!

**Comment by IzzySoft** [comment]
PS: the APK at tag `v0.3.0` identifies as `0.2.0` (see above). Looks like you misnamed the tag?

**Comment by freshfieldreds** [comment]
Thank you for notifying me about this. I didn't realise the last release had that inaccuracy, I'll make sure the future releases are correctly versioned.

**Comment by freshfieldreds** [comment]
I have released a new version of the app with the versioning issue fixed. You should be able to safely set it back to daily now.

**Comment by IzzySoft** [comment]
Partly confirmed:

```
$ iod repo get com.freshfieldreds.muffed
com.freshfieldreds.muffed: looking for 'https://api.github.com/repos/freshfieldreds/muffed/releases'
com.freshfieldreds.muffed: checking tag 'v0.4.0'
com.freshfieldreds.muffed: lastRelNo set to '0.4.0', checking for files
com.freshfieldreds.muffed: Upstream file date (2023-10-11 11:08) is newer than ours (2023-10-10 09:34).
com.freshfieldreds.muffed: returning ['0.4.0','https://github.com/freshfieldreds/muffed/releases/download/v0.4.0/muffed-v0.4.0.apk',1697015288]
com.freshfieldreds.muffed: 0.2.0/0.4.0, https://github.com/freshfieldreds/muffed/releases: https://github.com/freshfieldreds/muffed/releases/download/v0.4.0/muffed-v0.4.0.apk
- Grabbing update for com.freshfieldreds.muffed: OK
Binary files /web/ftp/repo/fdroid/repo/com.freshfieldreds.muffed_1.apk and /web/ftp/repo/fdroid/repo/com.freshfieldreds.muffed_0.4.0.apk differ
repo/com.freshfieldreds.muffed_1.apk already exists, quarantining before replacing it with update '/web/ftp/repo/fdroid/repo/com.freshfieldreds.muffed_0.4.0.apk'
- Checking 'repo/com.freshfieldreds.muffed_1.apk' for libraries and malware …
com.freshfieldreds.muffed: check if repo contains FUNDING.yml
com.freshfieldreds.muffed: looking for 'https://api.github.com/repos/freshfieldreds/muffed/contents/.github'
com.freshfieldreds.muffed: Github reports "Not Found" for https://api.github.com/repos/freshfieldreds/muffed/contents/.github
com.freshfieldreds.muffed: looking for 'https://api.github.com/repos/freshfieldreds/muffed/contents/'
com.freshfieldreds.muffed: no FUNDING.yml detected.
com.freshfieldreds.muffed: no Fastlane configured, skipping Fastlane check.
```

You still forgot to add the `versionCode` [here](https://github.com/freshfieldreds/muffed/blob/main/pubspec.yaml#L6) (`version: 0.4.0+4`, the `+4` is missing) – so the APK again got `versionCode: 1`. This means, whoever had the app already installed won't receive update notifications, nor an update option (as Android considers it the very same version as installed). So I've set the update check to "daily" now (as the tag name corresponds to `versionName`), but you still need to fix the `versionCode` for future releases so people can properly get notified and can update.

* `no FUNDING.yml detected`: you can ignore this message unless you want to publish donation infos. See [FUNDING.yml](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/displaying-a-sponsor-button-in-your-repository) for details.
* `no Fastlane configured, skipping Fastlane check`: Fastlane offers you a way to maintain your app description, changelogs and graphics. I'd recommend you to establish those structures here (be welcome to my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) for guidance). If you want me to, I can send you a "starter package" via PR, with what I've set up here for your app.

**Comment by freshfieldreds** [comment]
Thank you for all of your help, I have added the fastlane folder and will make sure the future versions are correct

**Comment by IzzySoft** [comment]
Thanks! A little note: Your `full_description.txt` should not just be a copy of `short_description.txt` but be a little more detailed so folks get an idea what to expect. Most people are less interested in what framework you used for programming ("written in flutter") but rather want to know what the app offers (features etc). So I'd suggest `a Lemmy client with a focus on usability and privacy` as `short_description.txt`, plus something more for the `full_description.txt`:

```markdown
<p><b>Muffed</b> is a Lemmy client written in flutter with a focus on usability and privacy. It lets you xxx and zzz. Current features highlights include:</p><ul><li>this</li><li>that</i></ul>
```

(I'm using HTML here as the output looks better, compressed to a single line to keep F-Droid clients from converting newlines to `<br>`).

I've enabled pulling the screenshots along with new releases now, looking forward to enable the descriptions when you decided for their "final state" (of course you can update them anytime later, e.g. when new features were added).

**Comment by freshfieldreds** [comment]
Thank you for your advice i have updated the descriptions

**Comment by IzzySoft** [comment]
Thanks! Enabling the pull of it then.

Just a last hint: While using headings (`<h1>` etc) is possible, it's a bad idea to use anything above `<h3>` (I'd even say only use `<h4>` and lower if you really need that). Headings are less for formatting but more about semantics. So higher headings might be used by websites themselves, e.g. for the app name (`<h1>`) and sections like "Description" or "Screenshots" (`<h2>`). There might be a level above that even (site name, or "Apps") which would shift that another level, hence my suggestion to at max start with `<h4>`. The `id`s won't be used anyway (they are not referred to by anything), so you could simply skip them. Further, some introductory text is missing.

That said, and for that reason, my updater removes them, replacing them with bold (and the first 3 "higher levels" with bold+bigger). F-Droid.org does not, so your listing there might look a bit weird with the current formatting (plus the fact that the F-Droid client replaces all line breaks with `<br>`).

```markdown
<p><b>Muffed</b> is a Lemmy client written in flutter with a focus on usability and privacy. <a href='https://github.com/LemmyNet/lemmy'>Lemmy</a> is a decentralised reddit alternative in the Fediverse, using the Activity-pub protocol – and thus can communicate with other Fediverse services.</p><h4>Links</h4><ul><li><a href="https://github.com/freshfieldreds/muffed">Github repo</a></li><li><a href="https://sh.itjust.works/c/muffed">Lemmy Community</a></li></ul><h4>Features</h4>
```

This (complemented with the rest of the text in the same manner) is what I would suggest.

**Comment by freshfieldreds** [comment]
When I released 0.8.5 the APK failed to upload which I didn't notice until a few days after I submitted the release when I got told about the issue. Probably part of the reason why it is not showing correctly.

**Comment by IzzySoft** [comment]
F-Droid.org checks the source, as it builds from source (it clones the entire repo and operates on its code). My repo is a binary repo, it wouldn't know how/where to find the corresponding values (as depending on the programming language used and other things, it can be in several different places and formats); my updater only pulls the APKs, not the entire repo. So I must be able to rely on the devs here to stick to the naming schemes they've chosen. I think that's fair enough. Mistakes happen on both ends, so the missing `v` in this case here was probably just an accidental slip.

Sorry that I cannot make everyone happy at the same time…
